pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2326,14 +2328,27 @@ public void checkInactiveSubscriptions() {
     @Override
     public void checkBackloggedCursors() {
         // activate caught up cursors which include consumers
+        AtomicReference<PositionImpl> slowestNonDurableReadPosition = new AtomicReference<>();
         subscriptions.forEach((subName, subscription) -> {
+            ManagedCursor cursor = subscription.getCursor();
+            if (cursor instanceof NonDurableCursorImpl) {
+                PositionImpl readPosition = (PositionImpl) cursor.getReadPosition();
+                if (slowestNonDurableReadPosition.get() == null || readPosition.compareTo(
+                        slowestNonDurableReadPosition.get()) < 0) {
+                    slowestNonDurableReadPosition.set(readPosition);
+                }
+            }
             if (!subscription.getConsumers().isEmpty()
                 && subscription.getCursor().getNumberOfEntries() < backloggedCursorThresholdEntries) {
                 subscription.getCursor().setActive();
             } else {
                 subscription.getCursor().setInactive();
             }
         });
+        if (slowestNonDurableReadPosition.get() != null) {
+            ManagedLedger managedLedger = getManagedLedger();
```

Comment:
You can directly use `ledger.updateTheSlowestNonDurableReadPosition`

Reply:
+1

Reply:
Agrre, and there will reset `ManagedLedgerImpl.slowestNonDurableReadPosition` when there is no non-duranble cursor.